tests/cases/compiler/noImplicitAnyDestructuringVarDeclaration2.ts

```diff
@@ -0,0 +1,12 @@
+// @noimplicitany: true
+let [a, b, c] = [1, 2, 3]; // no error
+let [a1 = 10, b1 = 10, c1 = 10] = [1, 2, 3]; // no error
+let [a2 = undefined, b2 = undefined, c2 = undefined] = [1, 2, 3]; // no error

```

Comment:
Should having default initializer be "undefined" get an noImplicitAny error? as it is similar to doing this 
``` var x = undefined``` or ``` function foo(x=undefined) {}``` in both case, the compiler will give an error